### PR TITLE
`General`: Fix an issue with routing URLs outside of the main web page

### LINKED ISSUE #4504: `Download PDF` button not working
### Describe the bug

On `Lectures`, when I try to download the merged PDF file by clicking `Download PDF`, it does not work.
Getting individual PDF files works, but getting a merged PDF file does not work.
Considering it started happens recently, it might be related to the recent releases.

### To Reproduce

1. Go to a random lecture.
2. Click on `Download PDF` on the top.

### Expected behavior

It should show up a merged PDF.

### Screenshots

_No response_

### What browsers are you seeing the problem on?

Chrome, Safari, Microsoft Edge, Firefox

### Additional context

_No response_

### Relevant log output

```shell
ERROR Error: Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'api/files/attachments/lecture/281/merge-pdf'
```

### PR DESCRIPTION
**Pull request by MerlinMehmed**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

The service worker interferes with some of the server routes causing issues. An example is 
#4504 where the user cannot open the merged PDF file of all lecture's PDFs. Therefore, we need to define routes that the service worked should exclude. 

### Description

This PR defines the routes in the service worker configuration that belong to the server application.

Fixes #4504

Fixes #4506 

### Steps for Testing

Prerequisites:
1. Log in to Artemis
2. Navigate to Course Administration
3. Find a lecture with at least one PDF file unit
4. Click "Download PDF" 
5. You should be able to access the merged PDF file

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by dveljacic — APPROVED** [review]
Tested on TS2. I was able to download and open merged PDF, as well as the individual PDFs. 👍

**Review by Strohgelaender — APPROVED** [review]
Tested on TS2, can confirm that downloading merged pdf files works again.

File: ngsw-config.json [hunk 1]
```diff
@@ -17,5 +17,16 @@
         "files": ["/content/**", "/*.(eot|svg|cur|jpg|png|webp|gif|otf|ttf|woff|woff2|ani)"]
       }
     }
+  ],
+  "navigationUrls": [
+    "!/**/*.*",
```
Comment (dfuchss) [inline ngsw-config.json:22]: Maybe `'/**',` is missing.
See https://angular.io/guide/service-worker-config#matching-navigation-request-urls

Pattern: Include anything and then specific exclusions.

**Review by dfuchss — CHANGES_REQUESTED** [review]

File: ngsw-config.json (hunk 1 above)
Comment (dfuchss) [reply ngsw-config.json:22]: But if it works, this may be not the case.

Comment (MerlinMehmed) [reply ngsw-config.json:22]: I tried both and didn't find a difference. That's also why I removed it

**Review by valeriebucher — APPROVED** [review]
Accessing merged PDF files worked for me!

File: ngsw-config.json (hunk 1 above)
Comment (dfuchss) [reply ngsw-config.json:22]: Ok. I have no strong opinion on whether this shall be included or not

**Review by dfuchss — APPROVED** [review]
Code looks good & Works on our test systems (SAML Feature tested).

**Review by baranimo7 — APPROVED** [review]
Tested on ts1. Downloading files work as expected.

**Review by coaxial1 — APPROVED** [review]
Tested on TS1. Seems to work fine without ``/**`` included even though the default suggests that it's needed 👍

**Review by krusche — APPROVED** [review]

**Pull request merged**